docs/style.md

```diff
@@ -125,3 +126,89 @@ While convenient for editing, it may be poorly legible in unrendered form
 <!-- BAD -->
 Definitely don't use soft-wrapping. While markdown mostly ignores source level line breaks, relying on soft wrap makes the source completely unreadable, especially on modern wide displays.
 ```
+
+## [Tracing](https://tracing.rs)
+
+When emitting events and spans with `tracing` prefer adding variable data via
+[`tracing`'s field mechanism][fields].
+
+[fields]: https://docs.rs/tracing/latest/tracing/#recording-fields
+
+Most apparent violation of this rule will be when the message utilizes any form
+of formatting, as seen in the following example:
+
+```rust
+debug!(
+    target: "client",
+    "{:?} Received block {} <- {} at {} from {}, requested: {}",
+    self.client.validator_signer.as_ref().map(|vs| vs.validator_id()),
+    hash,
+    block.header().prev_hash(),
+    block.header().height(),
+    peer_id,
+    was_requested
+);
+```
+
+Most of the time an event will not need to format anything into the message text
+in order to pass more information:
+
+```rust
+debug!(
+    target: "client",
+    message = "Received block",
+    validator_id = self.client.validator_signer.as_ref().map(|vs| {
+        tracing::field::display(vs.validator_id())
+    }),
+    %hash,
+    "block.previous_hash" = %block.header().prev_hash(),
+    "block.height" = block.header().height(),
+    %peer_id,
+    was_requested
+);
+```
+
+Always specify the `target` explicitly.
```

Comment:
We should provide more guidance on how to pick a value for the `target`. Naming it after a crate in most cases makes sense, but sometimes it may make sense to diverge. We should also provide some example `target`s...

Reply:
Yeah... The current rule is the minimal rule which keeps the code consistent with *pre-existing* practice.